sources/web/datalab/static/notebook-app.js

```diff
@@ -95,6 +95,9 @@ define(['static/appbar', 'static/minitoolbar'], function(appbar, minitoolbar) {
       })
     });
 
+    idleTimeout.setupKernelBusyHeartbeat();
+    $('#mainArea').scroll(idleTimeout.notebookScrolled);
```

Comment:
I haven't played around as much with non-notebook pages. The list page is very reset-happy, it seems to refresh the list of notebooks pretty often, so I didn't bother adding anything else there.